### ISSUE TITLE
Is it possible to make dark mode / light mode appropriate tray icons on windows?

### ISSUE THREAD

**Issue by scottjg** [electron/electron#24335]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description
For the native tray icons on windows, switching to dark mode will invert them. This is very similar to the behavior on macOS (though on macOS it seems to work for third party apps as well, including electron apps). As far as I can tell, there isn't a clear way for an electron app to do a tray icon on windows that automatically inverts when the light/dark mode setting switches.

I think part of the issue is that the app mode vs the shell mode is two settings, and electron doesn't expose the shell mode: https://github.com/electron/electron/issues/19729

### Proposed Solution
Perhaps electron could provide a tray api update that lets you specify alternative tray icons for light and dark mode on windows?

### Alternatives Considered
At the very least, an api that exposes which mode the shell is in, could be nice. Not sure how you'd auto update the tray icon without polling, though. is there a window message or some kind of event we could listen on for this?

cc @hacdias @deermichel

**Comment by ahallicks** [comment]
I can't find anything that automatically does this but you should be able to use https://www.electronjs.org/docs/api/native-theme#event-updated to check for changes to the native theme. You can then use some of the properties of nativeTheme to check for what the current system is using.

**Comment by hacdias** [comment]
The native theme API allows us to know the preferred _app_ colors scheme and not the OS colors scheme. I believe that this issue depends on #19729.

**Comment by panther7** [comment]
+1 for add "systemTheme"

**Comment by panther7** [comment]
Javascript solution:
```js
import { nativeTheme } from 'electron';
import Registry from 'winreg';

async function isUsedSystemLightTheme() {
    const reg = new Registry({
        hive: Registry.HKCU,
        key: '\\Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize',
    });
    return new Promise((resolve) => reg.get('SystemUsesLightTheme', (err, item) => {
        if (!err) return resolve(item.value === '0x1');
        resolve(false);
    }));
}

nativeTheme.on('updated', () => console.log(isUsedSystemLightTheme()) );
```

**Comment by MarshallOfSound** [comment]
Duplicate https://github.com/electron/electron/issues/19729

**Comment by panther7** [comment]
fixed in #46429
